indico/modules/events/editing/controllers/backend/editable_list.py

```diff
@@ -0,0 +1,80 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2020 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+import uuid
+
+from flask import request, session
+from werkzeug.exceptions import Forbidden
+
+from indico.legacy.common.cache import GenericCache
+from indico.modules.events.editing.controllers.base import RHEditablesBase, RHEditableTypeManagementBase
+from indico.modules.events.editing.models.editable import Editable
+from indico.modules.events.editing.operations import assign_editor, generate_editables_zip, unassign_editor
+from indico.util.i18n import _
+from indico.util.marshmallow import Principal
+from indico.web.args import use_kwargs
+from indico.web.flask.util import url_for
+from indico.web.util import jsonify_data
+
+
+archive_cache = GenericCache('editables-archive')
+
+
+class RHPrepareEditablesArchive(RHEditablesBase):
+    def _process(self):
+        key = unicode(uuid.uuid4())
+        data = [editable.id for editable in self.editables]
+        archive_cache.set(key, data, time=1800)
+        download_url = url_for('.download_archive', self.event, type=self.editable_type.name, uuid=key)
+        return jsonify_data(flash=False, download_url=download_url)
```

Comment:
`jsonify(download_url=download_url)` (`jsonify_data` is only needed for the old jquery-based stuff that expects a `success: true` field in the json)